src/blr/cli/tagfastq.py

```diff
@@ -119,12 +121,12 @@ def parse_corrected_barcodes(open_file):
     return corrected_barcodes
 
 
-def parse_raw_barcodes(open_file):
+def parse_uncorrected_barcodes(open_file):
     """
     Parse FASTA/FASTQ containing barcode sequences and return a dictionary with entry headers
     pointing to a raw barcodes sequence.
     :param open_file: dnaio odject.
-    :return: dict: entry headers pointing to a raw barcodes sequence
+    :return: iterable with dict elements
     """
     for barcode in tqdm(open_file, desc="Raw barcodes processed"):
```

Comment:
Doesn’t this `tqdm` now interfere with the other progress report in line 61?

Reply:
Not really, check this output from the latest run:

```
tagfastq - INFO: Starting
Clusters processed: 7774460it [00:21, 356120.73it/s]
tagfastq - INFO: Input detected as paired FASTQ.
tagfastq - INFO: Output detected as paired FASTQ.
Read pairs processed: 437146385it [2:11:42, 55314.71it/s]
tagfastq - INFO: Read-pairs missing barcodes: 4030880      
tagfastq - INFO: Finished
Raw barcodes processed: 433168106it [2:11:44, 54801.86it/s]
```

The buffer from "Read pairs processed" and "Raw barcodes processed" appear next to each other on separate lines during the run. After the run however the "Raw barcodes processed" is move below the other outputs, possibly as the iterator is not formally exhausted.

Reply:
Ok, I guess tqdm is smart about this. Good!